packages/react/src/components/DataGrid/index.tsx

```diff
@@ -0,0 +1,241 @@
+                                    tabIndex: 0,
+                                    onClick:
+                                      header.column.getToggleSortingHandler(),
+                                  }
+                                : {})}
+                            >
+                              {flexRender(
+                                header.column.columnDef.header,
+                                header.getContext()
+                              )}
+                              {header.column.getIsSorted() === "asc" && (
+                                <span className="material-icons">
+                                  arrow_drop_up
+                                </span>
+                              )}
+                              {header.column.getIsSorted() === "desc" && (
+                                <span className="material-icons">
+                                  arrow_drop_down
+                                </span>
+                              )}
+                              {!header.column.getIsSorted() && (
+                                <span className="c__datagrid__header__icon-placeholder" />
+                              )}
+                            </div>
+                          )}
+                        </th>
+                      );
+                    })}
+                  </tr>
+                ))}
+              </thead>
+              <tbody>
+                {table.getRowModel().rows.map((row) => (
+                  <tr key={row.id} data-testid={row.id}>
+                    {row.getVisibleCells().map((cell, i) => {
+                      let highlight = false;
+                      if (enableRowSelection && i > 0) {
+                        // Enabling selection adds a column at the beginning of the table.
+                        highlight = !!columns[i - 1].highlight;
+                      } else {
+                        highlight = !!columns[i].highlight;
+                      }
+                      return (
+                        <td
+                          key={cell.id}
+                          className={classNames({
+                            "c__datagrid__row__cell--highlight": highlight,
+                          })}
+                        >
+                          {flexRender(
+                            cell.column.columnDef.cell,
+                            cell.getContext()
+                          )}
+                        </td>
+                      );
+                    })}
+                  </tr>
+                ))}
+              </tbody>
+            </table>
+            {!!pagination && <Pagination {...pagination} />}
+          </>
+        )}
+      </>
+    );
+  };
+
+  return (
+    <div
+      className={classNames("c__datagrid", {
+        "c__datagrid--empty": isEmpty,
+        "c__datagrid--loading": isLoading,
+      })}
+    >
+      {getContent()}
```

Comment:
Why not create a sub component? This syntax is not really "reactish"

Reply:
Because with this approach we can use a guard pattern by returning early when `showEmptyPlaceholder` is true. Furthermore, create a sub component would force use to pass again all the vars used in `getContent` which are .. a lot, thus creating more verbosity 😕